### PR TITLE
Create glossary.md

### PR DESCRIPTION
**Pull request by beckettsean**
partial fix for #542

### CODE REVIEW

**Comment by beckettsean** [comment]
@sparrc @kostasb the glossary page is ready for some details. Please add whatever you can, don't worry about the language. Regan and I can clean it up.

File: content/telegraf/v1.1/concepts/glossary.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Related entries: [output plugin](/telegraf/v1.1/concepts/glossary/#output-plugin)
+
+## output plugin
+
+Output plugins deliver metrics to their configured destination. In order to activate an output plugin, it needs to be enabled and configured in the corresponding section in the given config file.
+
+Related entries: [aggregator plugin](/telegraf/v1.1/concepts/glossary/#aggregator-plugin), [flush interval](/telegraf/v1.1/concepts/glossary/#flush-interval), [input plugin](/telegraf/v1.1/concepts/glossary/#input-plugin), [processor plugin](/telegraf/v1.1/concepts/glossary/#processor-plugin)
+
+## precision
```
Comment (beckettsean) [inline content/telegraf/v1.1/concepts/glossary.md]: @kostasb does this control the precision of the emitted metrics? Would it change the timestamps passed to aggregator or processor plugins?

File: content/telegraf/v1.1/concepts/glossary.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+## processor plugin 
+
+Processor Plugins transform, decorate, and/or filter metrics and run in between input and output plugins.
+
+Related entries: [aggregator plugin](/telegraf/v1.1/concepts/glossary/#aggregator-plugin), [input plugin](/telegraf/v1.1/concepts/glossary/#input-plugin), [output plugin](/telegraf/v1.1/concepts/glossary/#output-plugin)
+
+## service input plugin
+
+Service input plugins are input plugins that run in a passive collection mode while the Telegraf agent is running. 
+Service input plugins listen on a socket for known protocol inputs, or actively apply their own logic to collect metrics and deliver them to the Telegraf agent at will.
```
Comment (beckettsean) [inline content/telegraf/v1.1/concepts/glossary.md]: So are service input plugins passive or not? Can you point me to examples of ones that are active? Why aren't they just input plugins?

File: content/telegraf/v1.1/concepts/glossary.md (hunk 1 above)
Comment (kostasb) [inline content/telegraf/v1.1/concepts/glossary.md]: I got this from some code comments, however after looking at the code it seems to me that precision is always set to nanoseconds. Modifying this parameter does not seem to affect the points precision. We need feedback from @sparrc for this.

File: content/telegraf/v1.1/concepts/glossary.md (hunk 2 above)
Comment (sparrc) [inline content/telegraf/v1.1/concepts/glossary.md]: All are passive servers, such tcp_listener, http_listener, udp_listener, kafka_consumer, mqtt_consumer, statsd, etc.

Comment (kostasb) [inline content/telegraf/v1.1/concepts/glossary.md]: They are passive in a sense that they all just sit there and wait for inputs. The word "actively" doesn't fit there, my bad.

File: content/telegraf/v1.1/concepts/glossary.md (hunk 1 above)
Comment (kostasb) [inline content/telegraf/v1.1/concepts/glossary.md]: So the timestamp's precision is truncated but write precision remains in ns.

E.g. 1479295580998402598 will be truncated to 1479295580 and then filled with trailing 0's to reach ns precision again as 1479295580000000000

Comment (kostasb) [inline content/telegraf/v1.1/concepts/glossary.md]: @sparrc just to clarify, I see that the accumulator sets the precision on per-plugin basis, so input plugins deliver their data through Gather to the agent using the configured precision. 
So it changes the timestamp before points are passed to aggregator or processor plugins.

Is this correct?

Comment (sparrc) [inline content/telegraf/v1.1/concepts/glossary.md]: @kostasb yes, that is correct

Comment (beckettsean) [inline content/telegraf/v1.1/concepts/glossary.md]: @sparrc does this commit seem correct, then? https://github.com/influxdata/docs.influxdata.com/pull/866/commits/2fcfb9fff87917e6ac2e6b09e48a4eb33ac29ecb

**Comment by kostasb** [comment]
After quite some digging in to the indexing order, I think this is ready to get a review & merge @rkuchan !

**Comment by rkuchan** [comment]
@kostasb - weights look good :)

I made a couple small changes. If you approve of them I think this is good to go. I did change `output plugins must emit all timestamps` to `input plugins must emit all timestamps` in the definition of precision. I think that's right but feel free to change it back if my understanding is wrong.

**Comment by kostasb** [comment]
@rkuchan thanks for pointing this out, I guess it was unclearly written
output plugins only support nanoseconds precision. input plugins use the configurable precision setting to truncate timestamps to the desired precision, but telegraf will then pad the timestamp with zeros to bring it back to ns precision 

Please feel free to modify the precision section if you feel that the wording doesn't feel right

**Comment by rkuchan** [comment]
@kostasb - got it. Thanks!

**Pull request merged**